app/main/models/models.py

```diff
@@ -58,6 +58,52 @@ def format(self):
         }
 
 
+class Cluster(BaseModel):
+    """
+    Represents a specific cluster, which is a grouping of nodes.
+    """
+    __tablename__ = 'clusters'
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String, nullable=False)
+    #represents the cartesian coordinates of the center of the cluster
+    x_position = db.Column(db.Integer)
+    y_position = db.Column(db.Integer)
+    # The project that holds this source
+    project_id = db.Column(db.Integer, db.ForeignKey('projects.id'),
```

Comment:
Since you're making a reference to project, you also need to add some stuff to the `Project` class. It would need a `clusters` property similar to the `sources` property.